tests/test_notifier.py

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/python3 -OO
+# Copyright 2007-2024 by The SABnzbd-Team (sabnzbd.org)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+"""
+tests.test_notifier - Testing notification functionality
+"""
+import sys
+from sabnzbd.notifier import send_apprise, NOTIFICATION_TYPES
+from unittest import mock
+
+if sys.version_info[0] == 3 and sys.version_info[1] < 11:
```

Comment:
What is the difference between the 2 tests?

Reply:
`@mock.patch` in Python v3.11 has different behavior then earlier versions. `apprise.Apprise.add` wouldn't work as it does in previous versions.

The only workaround i found was to do the exact same test with the `@mock.patch` handling for pre-py311 and post-py311. Post versions needed `apprise.Apprise.Apprise.add` (which doesn't with the earlier versions)

The `unittest.skipIf` doesn't work in Python 3.12; it kept running the pre-py311 test which failed because of the `@mock.patch`

I also tried doing the same if block as this and just swapped the strings to be passed into the patch call, and that didn't work either (looked a whole lot cleaner). After 3.5hrs, i just got frustrated and what you saw worked so i just left it.

The problem is that Apprise works in all these environments, but the evolution of the `unittests` module has not been seamless between Python versions of the same major.

Perhaps in this case, there is something i need to do in my `apprise/__init__.py` to resume compatibility with Python 3.11 and forward for unit test references...